panic: runtime error: invalid memory address or nil pointer dereference

### OS 
```
Linux scan 5.3.0-1022-azure #23~18.04.1-Ubuntu SMP Mon May 11 11:55:56 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

### cmd
```
wc -l subdomain.txt
1969 subdomain.txt

cat subdomain.txt | httpx -silent -title -json -o result.json
```

### Get Error
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6faccc]

goroutine 517 [running]:
github.com/projectdiscovery/httpx/common/httpx.(*HTTPX).NewRequest(0xc000022b80, 0x7a8ff1, 0x3, 0xc000eb4570, 0x2c, 0xc000eb4570, 0x2c, 0x0)
        /root/go/pkg/mod/github.com/projectdiscovery/httpx@v0.0.1/common/httpx/httpx.go:151 +0x7c
main.analyze(0xc000022b80, 0x7a9e90, 0x5, 0xc000eb4510, 0x24, 0x0, 0xc00007d0e0, 0xc0000721e0)
        /root/go/pkg/mod/github.com/projectdiscovery/httpx@v0.0.1/cmd/httpx/httpx.go:165 +0x173
main.main.func2(0xc00000e0e0, 0xc000022b80, 0x7a9e90, 0x5, 0xc00007d0e0, 0xc0000721e0, 0xc000eb4510, 0x24)
        /root/go/pkg/mod/github.com/projectdiscovery/httpx@v0.0.1/cmd/httpx/httpx.go:121 +0xab
created by main.main
        /root/go/pkg/mod/github.com/projectdiscovery/httpx@v0.0.1/cmd/httpx/httpx.go:119 +0x88a
```

--- a/cmd/httpx/httpx.go
+++ b/cmd/httpx/httpx.go
@@ -207,9 +207,8 @@ retry:
 		builder.WriteString(fmt.Sprintf(" [%d]", resp.ContentLength))
 	}
 
-	title := ""
+	title := httpx.ExtractTitle(resp)
 	if scanopts.OutputTitle {
-		title = httpx.ExtractTitle(resp)
 		builder.WriteString(fmt.Sprintf(" [%s]", title))
 	}
 

--- a/common/httpx/httpx.go
+++ b/common/httpx/httpx.go
@@ -146,10 +146,12 @@ func (h *HTTPX) AddFilter(f Filter) {
 // NewRequest from url
 func (h *HTTPX) NewRequest(method, URL string) (req *retryablehttp.Request, err error) {
 	req, err = retryablehttp.NewRequest(method, URL, nil)
+	if err != nil {
+		return
+	}
 
 	// set default user agent
 	req.Header.Set("User-Agent", h.Options.DefaultUserAgent)
-
 	return
 }
 

